### PR TITLE
Qa/inc upgrade

### PR DESCRIPTION
**Pull request by manand31**
# Description of change
Added updates to mongodb collection for the below tests : 
test_mongodb_full_table.py, test_mongodb_full_table_id.py, test_mongodb_full_table_interruptible.py, test_mongodb_incremental.py .

# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_mongodb_full_table.py [hunk 1]
```diff
@@ -212,17 +212,24 @@ def test_run(self):
         #  -------------------------------------------
         #  ----------- Second full Table Sync ---------
         #  -------------------------------------------
-        # add 2 rows and run full table again, make sure we get initial number + 2
-
         with get_test_connection() as client:
+        # update existing documents in the collection to make sure we get the updates as well in the next sync
```
Comment (kspeer825) [inline tests/test_mongodb_full_table.py:216]: these comments are not indented to the right level

File: tests/test_mongodb_full_table.py [hunk 2]
```diff
@@ -275,13 +282,14 @@ def test_run(self):
             self.assertNotEqual(first_versions[tap_stream_id], second_versions[tap_stream_id])
 
             # version which is larger than the previous target version
-            self.assertTrue(second_versions[tap_stream_id]>first_versions[tap_stream_id])
+            self.assertTrue(second_versions[tap_stream_id] > first_versions[tap_stream_id])
 
             # verify that menagerie state does include the version which matches the target version
             self.assertEqual(records_by_stream[self.tap_stream_id_to_stream()[tap_stream_id]]['table_version'], second_versions[tap_stream_id])
 
+            # TODO: check with kyle on the below two assertions, they are identical to the above two assertions
```
Comment (kspeer825) [inline tests/test_mongodb_full_table.py:290]: Yeah these are no necessary if they are deplicates. Also we should remove the assertTrue and use assertGreater

File: tests/test_mongodb_full_table_interruptible.py [hunk 3]
```diff
@@ -181,6 +198,28 @@ def test_run(self):
...
+            # We are not considering the last element of this list because it does not have 'data'
+            if int(x['data']['int_field']) == 999:
+                int_value = True
+        self.assertEqual(False, int_value)
+
+        int_value2 = False
+        for x in records_by_stream['simple_coll_1']['messages'][:-1]:
+            if x['data']['int_field'] == 888:
+                int_value2 = True
+        self.assertEqual(False, int_value2)
```
Comment (kspeer825) [inline tests/test_mongodb_full_table_interruptible.py:216]: Can you comment with what exactly this is checking. I am having a hard time following it.

File: tests/test_mongodb_incremental.py [hunk 4]
```diff
@@ -318,16 +394,56 @@ def test_run(self):
...
+            # Workaround for not including collections for uuid and string, TODO : look for a solution to implement string and uuid as replication_key
+            if k not in ('simple_coll_uuid_field', 'simple_coll_string_field'):
+                self.assertEqual(4, v)
 
         # Verify that the _id of the records sent are the same set as the
         # _ids of the documents changed
         for stream_name in self.expected_sync_streams():
-            actual = set([x['data']['int_field'] for x in records_by_stream[stream_name]])
-            self.assertEqual(self.expected_incremental_int_fields()[stream_name], actual)
+            # Workaround for not including collections for uuid and string, TODO : look for a solution to implement string and uuid as replication_key
```
Comment (kspeer825) [inline tests/test_mongodb_incremental.py:406]: Could you card this out as future work with label `tap-tester-upgrades1

File: tests/test_mongodb_incremental.py [hunk 5]
```diff
@@ -360,4 +476,4 @@ def test_run(self):
                                                                    self.expected_sync_streams(),
                                                                    self.expected_pks())
         for tap_stream_id in self.expected_sync_streams():
-            self.assertGreaterEqual(record_count_by_stream[tap_stream_id],self.expected_last_sync_row_counts()[tap_stream_id])
```
Comment (kspeer825) [inline tests/test_mongodb_incremental.py:363]: I think there should be an assertion here that validates we replicate all data for the collections

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tests/test_mongodb_full_table.py (hunk 1 above)
Comment (manand31) [reply tests/test_mongodb_full_table.py:216]: done

File: tests/test_mongodb_full_table.py (hunk 2 above)
Comment (manand31) [reply tests/test_mongodb_full_table.py:290]: done

File: tests/test_mongodb_full_table_interruptible.py (hunk 3 above)
Comment (manand31) [reply tests/test_mongodb_full_table_interruptible.py:216]: I have added relevant comments for others to understand

File: tests/test_mongodb_incremental.py (hunk 4 above)
Comment (manand31) [reply tests/test_mongodb_incremental.py:406]: https://jira.talendforge.org/browse/TDL-16915

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**